tests/comments/test_async_api_org_terms.py

```diff
@@ -0,0 +1,320 @@
+        assert response.data['comment'] == 'comment comment comment'
+
+    terms = OrganisationTermsOfUse.objects.all()
+    assert len(terms) == 0
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_comment_save_without_agreement_forbidden(
+        user, apiclient, question_ct, question):
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {
+        'comment': 'comment comment',
+        'agreed_terms_of_use': False
+    }
+    with active_phase(question.module, AskPhase):
+        response = apiclient.post(url, data, format='json')
+
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_comment_save_without_agreement_data_forbidden(
+        user, apiclient, question_ct, question):
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {
+        'comment': 'comment comment'
+    }
+    with active_phase(question.module, AskPhase):
+        response = apiclient.post(url, data, format='json')
+
+    assert response.status_code == status.HTTP_400_BAD_REQUEST
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_comment_save_with_already_agreed(
+        user, apiclient, question_ct, question,
+        organisation_terms_of_use_factory):
+    organisation_terms_of_use_factory(
+        user=user,
+        organisation=question.module.project.organisation,
+        has_agreed=True,
+    )
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {
+        'comment': 'comment comment',
+    }
+    with active_phase(question.module, AskPhase):
+        response = apiclient.post(url, data, format='json')
+
+    assert response.status_code == status.HTTP_201_CREATED
+    assert response.data['comment'] == 'comment comment'
+
+
+@pytest.mark.django_db
+@override_settings(
+    A4_USE_ORGANISATION_TERMS_OF_USE=True
+)
+def test_agreement_fields_not_implemented(
+        apiclient, question_ct,
+        question, organisation_terms_of_use_factory):
+    organisation_terms_of_use_factory(
+        user=question.creator,
+        organisation=question.module.project.organisation,
+        has_agreed=True,
+    )
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    data = {'comment': 'comment comment'}
+    with active_phase(question.module, AskPhase):
+        apiclient.post(url, data, format='json')
```

Comment:
Should this raise permission denied as user is not authenticated?

Reply:
Yeah, it is in there from other tests to add a comment and then test the info. But I am not actually using it, so it can just be removed. Same in the next test. Will do!